src/containers/Settings.jsx

```diff
@@ -143,6 +148,85 @@ class Settings extends React.Component {
     );
   }
 
+  handleSubmitTwilioAuthForm = async ({
+    apiKey,
+    authToken,
+    messageServiceSid
+  }) => {
+    await this.props.mutations.updateTwilioAuth(
+      apiKey,
+      authToken === '<Encrypted>' ? false : authToken,
+      messageServiceSid
+    );
+  };
+
+  renderTwilioAuthForm() {
```

Comment:
I'm concerned that allowing all fields to be updated after the organization is set up leaves a lot of room for misconfiguration. Maybe add a confirmation step if someone updates a per-existing Twilio Account SID or Messaging Service since updating either will break all existing campaigns? I think it's probably worth throwing in a simple twilio request in the update mutation to validate that the settings are correct too.

Another approach to make changing configuration easier would be to preserve the previous settings for existing campaigns. We did that for Messaging Services in the Warren fork to facilitate migrating from global to org-wide and finally to campaign-specific messaging services. The sid is written to the campaign table in [start_campaign](https://github.com/Elizabeth-Warren/Spoke/blob/master/src/server/workers/start-campaign.js#L63-L78) and used in [send message](https://github.com/Elizabeth-Warren/Spoke/blob/master/src/server/api/lib/twilio.js#L94-L105). I think we should do the same when implementing per campaign messaging services (#1495) cc/ @hiemanshu. It's probably overkill to store auth settings since I would expect the account sid to change infrequently once the org is set up.

Reply:
Another option is the way they're doing it in the rewired fork — they have a `messaging_service` table and look up the account SID/token based on the campaign-configured messaging service.

That felt like overkill to me, and isn't easily backwards compatible. I think a warning is sufficient, and a good idea.

Reply:
Defer to @matteosb on the table (and what to do here, vs later changes). Whatever we do, let's note that it IS possible to rotate credentials, so there's at least some case where you might want to update creds (at least the SECRET) on an existing messageservicesid

Reply:
@schuyler1d yes, absolutely, the changing the auth token should be fully supported and I don't think it requires a warning.

@jeffm2001 agree, i think something like the politics-rewired messaging service table would be a good solution if we want to support switching twilio accounts without breaking campaigns but it's a bit overkill and makes migration more complicated.